### PR TITLE
Adds DD_TRACE_AGENT_TIMEOUT_SECONDS

### PR DESCRIPTION
**Pull request by sonuGlobality**
## Description
For sending apm metrics to localhost, the defaults of `2.0` is enough. However, for setup where datadog agents are exposed over `https`, the timeout of 2.0 is not sufficient. Such kind of setup is used for non-prod environment for devs to test their apm/stats configuration without having to install ddagent on every non-prod machines or generate multiple ddagent key for each dev to test themselves.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: ddtrace/internal/agent.py [hunk 1]
```diff
@@ -33,6 +33,11 @@ def get_stats_port():
     return int(get_env("dogstatsd", "port", default=DEFAULT_STATS_PORT))  # type: ignore[arg-type]
 
 
+def get_trace_url_timeout():
+    # type: () -> float
+    return float(os.environ.get("DD_TRACE_URL_TIMEOUT", default=DEFAULT_TIMEOUT))
```
Comment (brettlangdon) [inline ddtrace/internal/agent.py:38]:
```suggestion
    return float(get_env("trace", "url", "timeout", default=DEFAULT_TIMEOUT))
```

**Review by brettlangdon — COMMENTED** [review]
hey @sonuGlobality thanks for the PR!

Let me talk about this with others to confirm if we want to expose this (maybe there are others too?), and is there a specific env variable naming pattern we should use for it.

**Comment by sonuGlobality** [comment]
> hey @sonuGlobality thanks for the PR!
> 
> Let me talk about this with others to confirm if we want to expose this (maybe there are others too?), and is there a specific env variable naming pattern we should use for it.

Any update on this please?

**Comment by brettlangdon** [comment]
> Any update on this please?

@sonuGlobality yes, sorry for the delay. We are thinking a name like `DD_TRACE_AGENT_TIMEOUT_SECONDS` would be a better fit, but confirming with a few others first before we commit (we like to try and be consistent with env variable names across languages, if others adopt this we want to make sure everyone agrees on the naming).

**Comment by brettlangdon** [comment]
@sonuGlobality I made a few changes to the name, and added a release note. Let me know what you think?

**Comment by sonuGlobality** [comment]
> @sonuGlobality I made a few changes to the name, and added a release note. Let me know what you think?

Agree. It seems more consistent with other properties `DD_TRACE_AGENT_XXX` and addition of `SECONDS` leave no doubt on its unit

Awesome. Thank a ton guys for prompt actions on this PR.

**Review by brettlangdon — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]
👏 thanks @sonuGlobality!

**Pull request merged**